### DISCUSSION TITLE
Custom data attribute in select options?

### DISCUSSION THREAD

**Discussion by AbhinavNmdo** [filamentphp/filament#7831] [General]
Hello,

I wanted to know can I add a custom data attribute in the select options dynamically?

Thanks in advance

**Comment by pboivin** [comment]
You can use `->extraAttributes()` to add attributes on the Select tag, I'm not sure about the individual options... I'm curious, what's your use-case for this?

**Reply by AbhinavNmdo** [reply]
I wanted to set a value to an Hidden input, but after posting this I realize I am not able to access the data attribute for perticular option using `->afterStateUpdated()`.

Is there any way to get the whole collection in the callback of `->afterStateUpdated()`, because I want to use another db column to set the Hidden element. (Actually I don't want to run an another query to get the column).

**Reply by AbhinavNmdo** [reply]
This is an another quesction!!

![image](https://github.com/filamentphp/filament/assets/86583211/01dbb17e-b2da-4ab8-992c-ba99b137eaee)

I want to make an utility for the purchase items, whenever the value update in the `Purchase Items` section it will reflect in the `Purchase 
Total` section. And you know the logic behind the purchase module.

And the problem I facing here is I need to access it by using `$get('../../purchase_items')` parent, but when the repeater added it will continuously increasing the parent modifier. So, is there any solution?

**Reply by pboivin** [reply]
> Is there any way to get the whole collection in the callback of ->afterStateUpdated()

I don't think this is possible...
